### ISSUE TITLE
Barbarians cause Desync

### ISSUE THREAD

**Issue by enimog** [LoneGazebo/Community-Patch-DLL#2824]
_1. Mod version (i.e Date - 4/23):_
Latest

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
Not sure if it is really that, but when I tried to play in multiplayer, I had a desync after every ~4 turns or so. Turning off barbarians seemed to have fixed this issue.

**Comment by enimog** [comment]
Hello, it's me again.
I tried it again yesterday, using a modpack I made. It is possible that another mod I use cause the issue, but since https://forums.civfanatics.com/threads/how-to-interpret-logs-in-terms-of-desyncs.606457/ and other threads seems to mention this issue too, I think it is still worth investigating a little. 

I have included net messages from the client and the host for the first 3 turns (A desync happened turn 2, turn 1 and 3 were okay). Another fact I think might be important is that there is a huge difference in computing power between the host and the client I used. Don't know if this can change something. Seems to be something wrong with unit synchronization when looking at the logs.

[net_message_debug.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/623324/net_message_debug.zip)

**Comment by LoneGazebo** [comment]
@ilteroi thoughts?

**Comment by ilteroi** [comment]
we have 1000 lines of this, which is suspicious. there are desyncs already before, but that's the normal human player movement.

[21628.181] DBG: UpdateMoves() : player.AI_unitUpdate() called for player 63 Crom
[21628.306] DBG: UpdateMoves() : player.AI_unitUpdate() called for player 63 Crom
[21628.431] DBG: UpdateMoves() : player.AI_unitUpdate() called for player 63 Crom
[21749.175] DBG: UpdateMoves() : player.AI_unitUpdate() called for player 63 Crom
[21749.175] DBG: GAME HANG - Please show Ed and send save. Stuck units will have their turn ended so game can advance. [DETAILS: Player 63 Crom. First stuck unit is Incubator at (20, 11)]

what is an "Incubator"? does it have any specialties?

**Comment by supermerill** [comment]
same problem for me. Even with the 10-28 modpack #2848
So probably not an 'another mod' problem.

**Comment by LoneGazebo** [comment]
Interesting. All barbarian RNG calls are fake at this point, I believe.

**Comment by ilteroi** [comment]
can't reproduce the issue (in singleplayer at least)

**Comment by enimog** [comment]
@ilteroi Ah, About the Incubator, that one is most probably in the "another mod" bug territory so I don't really think you should mind about it, I wouldn't dare want you to fix somebody else mod :) 

But it is a unit that comes from the mod https://forums.civfanatics.com/threads/wip-madoka-magica-wish-for-the-world.543931/ that I slightly updated to fix bugs. It's base unit is a warrior. A quick search inside the whole mod project indicate that there is 2 lua functions related to it: One `GameEvents.PlayerDoTurn()`, where it creates the unit using `Players[iBarbarians]:InitUnit(iIncubatorType, pSpawnPlot:GetX(), pSpawnPlot:GetY(), UNITAI_ATTACK) ` near each player spawn on a RNG call each turn. The other function is a `GameEvents.UnitSetXY()` call type where it kill itself and update some table values upon specific conditions (being near a magical girl unit type). This work properly in SP.

If you are interested in using the exact same modpack as I am for debugging purpose, I took the time to upload it to mega. I deleted all sounds from it to reduce it's size from 4gb to 335mb. If you want to disable everything non-cp mod related, you could delete Lua/main.lua inside. https://mega.nz/#!ew5QRZxY!WcZ4bpOt_YfCsMNBDk-pHtuB2skSeYnY9FYS87EK3OU

**Comment by LoneGazebo** [comment]
@enimog I'm interested to see if you get desyncs in a 'standard' modpack.

**Comment by enimog** [comment]
Alright, I tried https://forums.civfanatics.com/threads/vox-populi-10-28-modpacks.603515/ with the same settings I used for that last game. After 20 turns, the only desync in the log file are the normal player interactions. Most noticeable thing is that there is no thousands of `UpdateMoves() : player.AI_unitUpdate()`. I'm tempted to say that this likely is a case of mod conflict/other mod bug after all and close the issue, but then again I'm not alone in this situation and after looking at @supermerill logs he has the same issue with the same debug messages for barbs. There might be something but I was not able to reproduce it this time...
